src/pages/UserSettingsPage/UserSettingsPage.tsx

```diff
@@ -0,0 +1,150 @@
+import Layout from 'src/components/Layout/Layout';
+import classes from './UserSettingsPage.module.scss';
+import { ExclamationCircleIcon } from '@heroicons/react/24/outline';
+import Message from 'src/components/Message/Message';
+import Button from 'src/components/core/Button/Button';
+import Input from 'src/components/core/Input/Input';
+import { useEffect, useState } from 'react';
+import { toast } from 'react-toastify';
+import { getUser, updateUser } from 'src/services/UserService';
+import { jwtDecode } from 'jwt-decode';
+import UpdateUser from 'src/models/UpdateUser';
+import Card from 'src/components/Card/Card';
+
+const UserSettinsPage: React.FC = () => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [validationErrors, setValidationErrors] = useState({
+    email: false,
+    firstName: false,
+    lastName: false,
+    username: false,
+    password: false,
+    phoneNumber: false
+  });
+  const [updatedUser, setUpdatedUser] = useState<UpdateUser>({
+    email: '',
+    firstName: '',
+    lastName: '',
+    username: '',
+    phoneNumber: ''
+  });
+
+  const id = jwtDecode(JSON.parse(localStorage.getItem('token')!)).id;
```

Comment:
This shouldn't be part of UserSettingsPage, please create decode token in AuthService